runtime/jvmti/jvmtiThreadGroup.c

```diff
@@ -205,38 +211,46 @@ jvmtiGetThreadGroupChildren(jvmtiEnv* env,
 				}
 			}
 
-            currentThread->javaVM->internalVMFunctions->objectMonitorExit(currentThread, childrenGroupsLock);
+#if defined(J9VM_OPT_OJDK_THREAD_SUPPORT)
+			currentThread->javaVM->internalVMFunctions->objectMonitorExit(currentThread, threadGroupObject);
+#else /* defined(J9VM_OPT_OJDK_THREAD_SUPPORT) */
+			currentThread->javaVM->internalVMFunctions->objectMonitorExit(currentThread, childrenGroupsLock);
+#endif /* defined(J9VM_OPT_OJDK_THREAD_SUPPORT) */
 
-            /* Construct the Children Threads array under a lock */
-			
+			/* Construct the Children Threads array under a lock */
+
+#if defined(J9VM_OPT_OJDK_THREAD_SUPPORT)
+			threadGroupObject = (j9object_t) currentThread->javaVM->internalVMFunctions->objectMonitorEnter(currentThread, threadGroupObject);
+			if (J9_OBJECT_MONITOR_ENTER_FAILED((UDATA)threadGroupObject)) {
+#else /* defined(J9VM_OPT_OJDK_THREAD_SUPPORT) */
 			threadGroupObject = *((j9object_t*) group);
```

Comment:
I was wrong about this moving, it should be before the #if since it's used in the #if case as well.

Reply:
Right, moved to the original location.